Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeCommon.azsli

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <viewsrg.srgi>
+
+#include <Atom/Features/SrgSemantics.azsli> 
+#include <Atom/Features/PostProcessing/FullscreenPixelInfo.azsli>
+#include <Atom/Features/PostProcessing/FullscreenVertex.azsli>
+
+partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
+{
+    //! The color buffer to apply the editor mode feedback effects to.
+    Texture2D<float4> m_framebuffer;
+
+    //! The non-linear depth buffer data for fragments in the color buffer.
+    Texture2D<float4> m_depth;
+
+    //! The entity mask containing the visible fragments of entities of interest.
+    Texture2D<float4> m_entityMask;
+
+    //! The final blend amount that is used to scale the calculated blend values.
+    float m_finalBlendAmount;
+    
+    Sampler LinearSampler
+    {
+        MinFilter = Linear;
+        MagFilter = Linear;
+        MipFilter = Linear;
+        AddressU = Clamp;
+        AddressV = Clamp;
+        AddressW = Clamp;
+    };
+
+    Sampler PointSampler
+    {
+        MinFilter = Point;
+        MagFilter = Point;
+        MipFilter = Point;
+        AddressU = Clamp;
+        AddressV = Clamp;
+        AddressW = Clamp;
+    };
+
+    //! Returns the dimensions of the specified 2D image.
+    float2 GetImageSize(Texture2D<float4> image)
+    {
+        float3 size = float3(0., 0., 0.);
+        image.GetDimensions(size.x, size.y);
+        return size;
+    }
```

Comment:
Why is the size a float3 and not a float2 here? Also this doesn't need to be in the PassSrg and can be moved outside and live as a standalone function. Even better though would probably be to directly use GetDimensions in the code so you don't need this function.

Reply:
That should indeed be a `float2`, not a `float3` (lazy copy and paste job by myself from a preexisting instance of this function) 👍  I'll check the usage of this function and see if it makes sense to put this logic in situ (or, at the very least, have it explicitly named for the mask texture)